src/utils.ts

```diff
@@ -21,67 +41,88 @@ const isSimilarTextExistInArray = (arr: string[], text: string) =>
  * This function returns all the nodes which are in the importOrder array.
  * The plugin considered these import nodes as local import declarations.
  */
-export const getSortedNodesByImportOrder = (
+export const getSortedNodes = (
     nodes: ImportDeclaration[],
     order: PrettierParserOptions['importOrder'],
+    importOrderSeparation: boolean,
 ) => {
-    return order.reduce(
-        (res: ImportDeclaration[][], val): ImportDeclaration[][] => {
+    const newLine =
+        importOrderSeparation && nodes.length > 1 ? newLineNode : null;
+
+    debugger;
+
+    const sortedNodesByImportOrder = order.reduce(
+        (
+            res: (ImportDeclaration | ExpressionStatement)[],
+            val,
+        ): (ImportDeclaration | ExpressionStatement)[] => {
             const x = nodes.filter(
                 (node) => node.source.value.match(new RegExp(val)) !== null,
             );
             if (x.length > 0) {
                 x.sort((a, b) => naturalSort(a.source.value, b.source.value));
-                return [...res, x];
+                return compact([...res, newLine, ...x]);
```

Comment:
Is it possible to have the newLine logic a bit more clear? Maybe a function can do that?